### ISSUE TITLE
HA can create its own device Matter credentials

### ISSUE THREAD

**Issue by barrelful** [home-assistant/android#4407]
**Is your feature request related to a problem? Please describe.**
Yes, today this code is managed by Google, and generates the error **Failed to generate device credentials** if you don't have a Google account in your Android device, which is confusing for the end user.

This was reported an discussed on #Thread Discord and also in this issue: https://github.com/home-assistant/core/issues/105535

**Describe the solution you'd like**
HA is able to generate the credentials by itself, this way this error would never occur and as an extra benefit a Google account dependency will not exist.

**Additional context**
It is a candidate feature I could try to give my first contribution to HA if someone is able to onboard me.

**Comment by fhmbrg** [comment]
It's a shame that the Companion App relies on Google Play services. How do people like me who are using GrapheneOS and don't want to install Play services/GMScompatConfig, add Matter devices to their Home Assistant.

Really would appreciate to have this working in my environment.

**Comment by adfrancis** [comment]
would love this!

**Comment by kayserlein** [comment]
Adding onto this. Having recently switched to GrapheneOS, it is frustrating to be limited here. Would really appreciate if something could be done here as it would truly round out the Matter experience on HA.

**Comment by CircuitGuy** [comment]
I looked into this briefly after being stymied and frustrated that (phone) internet and logged-in Google Play Services is required to commission a device.

Best I can - this is why:
Matter spec: jump to 6.2.3. Device Attestation Procedure, page 287
https://csa-iot.org/wp-content/uploads/2022/11/22-27349-001_Matter-1.0-Core-Specification.pdf

By spec, you have to validate that it's a genuine Matter device. They use blockchain for that.

Here's the document it eventually links you to after a few rabbit holes.
https://github.com/zigbee-alliance/distributed-compliance-ledger/blob/master/docs/transactions.md#how-to-read-from-the-ledger
-and finally a web UI-
https://webui.dcl.csa-iot.org/

The official Matter spec requires 'shall' inspect the ledger. After that, it's optional if you trust it - you have to warn the user.
Best I can tell, Google is pretty strict on compliance and won't offer to onboard an invalid (or counterfeit) device.

**_The whole dance looks pretty complicated, and devs are in a bit of a pickle here_** 
The gold standard would be for internet-connected HA to sync with this blockchain and validate devices semi-offline. I can't figure out the size of it easily, but it has 250k blocks as of now and so I would swag 250 MB. It'll still require connectivity of some sort, and again, it looks like an absolute PITA to actually implement.
 
To be certified to work with Matter, HA 'shall' validate the devices and warn the users if it's invalid. Not to mention it's really in our community best interest - counterfeit parts are a pain and would generate more issues than necessary. So it looks like either internet-connected Android through (free to use) Apple/Google nonsense, or quite a bit of dev work.

**Comment by jiriks74** [comment]
Yeah... I ran into this when I tried to pair my first M/T devices. Since I migrated my phone to GrapheneOS my Play Service don't have root privileges and so when it tries to get WiFi credentials through some method requiring root (AFAIK this is what it does), even though I'm connecting a Thread device, it fails and just says that I don't have a working internet connection.

Lucky me I have a cheap Android tabled with a camera so I was able to use that for pairing but the ability to pair using only HA only would be quite useful to many people I imagine.

From what others wrote here the spec does sound like pain and seems to be less open than it seems at a glance but it's still better than, sometimes multiple, manufacturer integrations that may or may not work and may or may not be stable.

**Comment by H3wastooshort** [comment]
There is now a good way to do this: Using the new JavaScript Matter server, with it's BLE-proxy endpoint enabled, one can use existing BLE proxies to pair new matter devices. Either from the "Settings -> Matter -> Options" section in homeassistant, or from the Matter Server WebUI. Tried it. Works fine.

Now how about integrating that into the "Add Matter Device" flow? Right now, one still has to copy-paste the QR-code content and the Thread TLV.

**Comment by Bodmoor** [comment]
+1 for not needing a google account to use matter over thread on my local network

**Comment by ForTheRush** [comment]
> **free to use** Apple/Google nonsense, or quite a bit of dev work.

Thats the thing though, those services aren't really **free** you are paying for them with your data, personal information, location data, habits, and pretty much ever descernable metric google can scrape from your life.  Sure you may not pay cash but ypu are absolutely paying them by using their services
